mealie/services/migrations/cookn.py

```diff
@@ -0,0 +1,409 @@
+            if pre_qualifier is not None:
+                og_text += pre_qualifier + " "
+            if food_name is not None:
+                og_text += food_name + " "
+            if post_qualifier is not None:
+                og_text += post_qualifier + " "
+            if brand is not None:
+                og_text += brand
+
+            base_ingredient = RecipeIngredient(
+                quantity=amount,
+                unit=unit,
+                food=food,
+                note=note,
+                original_text=og_text,
+                disable_amount=False,
+            )
+            try:
+                _display_order = db.get_data(_ingredient_row, "DISPLAY_ORDER")
+                ingredients_order.append(int(_display_order))
+                ingredients.append(base_ingredient)
+            except ValueError:
+                self.logger.warning("Invalid ingrediant order: %s, %s", _display_order, base_ingredient.original_text)
+                continue
+        return [obj for _, obj in sorted(zip(ingredients_order, ingredients, strict=False))]
+
+    def _parse_instructions(self, instructions: str) -> list[str]:
+        """
+        Parses recipe instructions into a list of steps.
+        Detects numbered lists, bulleted lists, and plain new-line-separated steps.
+        """
+        # Detects numbered lists (1., 1), 1-, etc.) and bulleted lists (-, *, •)
+        numbered_pattern = re.compile(r"^(\d+)[.)-]\s*(.*)")
+        bullet_pattern = re.compile(r"^[\-*•]\s*(.*)")
+
+        lines = instructions.splitlines()
+        steps = []
+        current_step: list[str] = []
+
+        for line in lines:
+            line = line.strip()
+
+            if not line:
+                continue  # Skip empty lines
+
+            num_match = numbered_pattern.match(line)
+            bullet_match = bullet_pattern.match(line)
+
+            if num_match:
+                # If there's a current step, store it before starting a new one
+                if current_step:
+                    steps.append("\n".join(current_step))
+                    current_step = []
+
+                current_step.append(num_match.group(2))
+            elif bullet_match:
+                if current_step:
+                    steps.append("\n".join(current_step))
+                    current_step = []
+
+                current_step.append(bullet_match.group(1))
+            else:
+                # Continuation of a previous step
+                if current_step:
+                    current_step.append(line)
+                else:
+                    # If no clear separator is found, treat each new line as a new step
+                    steps.append(line)
+
+        if current_step:
+            steps.append(" ".join(current_step))
+
+        return steps
+
+    def _process_recipe_document(
+        self, _recipe_row: dict[str, Any], db: DSVParser
+    ) -> dict[str, str | list[str] | list[RecipeIngredient]]:
```

Comment:
Even though this is technically correct, you can probably just leave off the extra typing info. More of a nit but it matches the other migrations.
```suggestion
    def _process_recipe_document(self, _recipe_row: dict[str, Any], db: DSVParser) -> dict:
```